crates/fuzz/src/program_test_client_blocking.rs

```diff
@@ -263,6 +269,53 @@ impl FuzzClient for ProgramTestClientBlocking {
             .block_on(self.ctx.banks_client.process_transaction(transaction))?)
     }
 
+    /// Process transaction, returning the result of the tx processing
+    /// in addition to `fn process_transaction()`, this function obtains transaction metadata
+    /// and adds CU units used by the transaction to the `stats_logger`
+    /// # Arguments
+    /// * `transaction` - transaction to process
+    /// * `instruction` - name of the instruction inside the transaction
+    /// * `stats_logger` - statistics logger
+    fn process_transaction_with_metadata(
+        &mut self,
+        transaction: impl Into<VersionedTransaction>,
+        instruction: String,
+        stats_logger: &mut FuzzingStatistics,
+    ) -> Result<(), FuzzClientError> {
+        let transaction_result = self.rt.block_on(
+            self.ctx
+                .banks_client
+                .process_transaction_with_metadata(transaction),
+        );
+
+        match transaction_result {
+            Ok(result_with_metadata) => {
+                // We got metadata => transaction was either successful or returned with TransactionError
+                match result_with_metadata.result {
```

Comment:
Personally, I would consider to first unwrap the Metadata and then based on the result (Ok , or Err), update the appropriate stats, but as I meantioned previously , this should be handled in executor not here.